src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 1 << 4;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= buckets.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = getIndex(key);
+        if (buckets[index] == null) {
+            buckets[index] = new Node<K, V>(key, value, null);
+        } else {
+            Node<K, V> currentNode = buckets[index];
+            while (currentNode != null) {
+                if (Objects.equals(key, currentNode.key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<K, V>(key, value, null);
+                    size++;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = buckets[index];
+        if (currentNode == null) {
+            return null;
+        }
```

Comment:
redundant if statement. 

in case currentNode is null, while loop won't start.

```suggestion
```